### PR TITLE
Academic reminders

### PR DESCRIPTION
**Pull request by sailoraria**
Implemented the rest of my use case up to fixing the json stuff (i just want to pull now in case it takes a while to fix the json, and to test my use case on the cloud to ensure everything works well with threading)

It can now: 
Send dm reminders to users (24 hrs before event or 1hr before event if the event is less than a day away)
Clear individual events from your schedule and unsubscribe you from the reminder alert
Remove passed events 
Unsubscribe from all reminder alerts if you clear the entire schedule

### CODE REVIEW

File: src/main/java/ca/unknown/bot/app/Main.java
Comment (Overpoweredplatypus) [inline src/main/java/ca/unknown/bot/app/Main.java:1]: Nice change. I don't think any reminder app uses second to determine whether an event passed.

File: src/main/java/ca/unknown/bot/data_access/schedule_reminder/ScheduledReminderDAO.java [hunk 1]
```diff
@@ -93,7 +114,7 @@ private Map<String, Schedule> load(String filename){
             // data which was written to it when saving the cache to a repo
             GsonBuilder builder = new GsonBuilder();
```
Comment (Overpoweredplatypus) [inline src/main/java/ca/unknown/bot/data_access/schedule_reminder/ScheduledReminderDAO.java:115]: Wouldn't Gsonbuilder be unnecessary if you are gonna be using new Gson();? Consider removing it.

**Review by Overpoweredplatypus — APPROVED** [review]
Wow, I can see how hard you've worked since the last meeting! Thank you so much :) Can't wait to try this out myself.

**Review by xinyue296 — CHANGES_REQUESTED** [review]
You've really put a lot of effort into this, great job! One thing to change is that time_respository.json should be deleted, so that it matches with the current main :) and also .json file should be added to gitignore for next commit!

**Review by xinyue296 — CHANGES_REQUESTED** [review]
I meant to edit my review, but accidentally re-requested myself as a reviewer instead :(
The changes you made are very nice! You also need to do a `git pull` since the `.gitignore` file on your branch is different from the current `origin main`.

**Review by xinyue296 — APPROVED** [review]

**Review by iDasher — APPROVED** [review]

**Pull request merged**